src/main/java/seedu/apollo/command/module/AddModuleCommand.java

```diff
@@ -228,6 +236,67 @@ public ArrayList<LessonType> getLessonTypes(Module module) {
         return lessonTypes;
     }
 
+    /**
+     * Checks if the lesson clashes with another lesson.
+     *
+     * @param calendar The calendar of the user containing timetable information.
+     * @param timetable The timetable of the lesson to be checked.
+     * @param ui The ui of the user for message printing.
+     */
+    private void checkClashingLesson(Calendar calendar, Timetable timetable, Ui ui) {
+        String day = timetable.getDay();
+        int index = determineDay(day);
+
+        if (index == -1) {
+            return;
+        }
+
+        if (calendar.get(index).size() == 0) {
+            return;
+        }
+
+        for (CalendarModule lessonModule: calendar.get(index)) {
+            Timetable schedule = lessonModule.getSchedule();
+            if (isClashing(schedule, timetable)) {
+                ui.printClashingLesson();
+                break;
+            }
+        }
+    }
+
+    /**
+     * Checks if a lesson clashes with another lesson.
+     *
+     * @param schedule The lesson to be checked.
+     * @param timetable The lesson to be checked against.
+     * @return True if the timetable clashes with another timetable.
+     */
+    private boolean isClashing(Timetable schedule, Timetable timetable) {
+
+        SimpleDateFormat format = new SimpleDateFormat("HHmm");
+        try {
+            Date start1 = format.parse(schedule.getStartTime());
+            Date start2 = format.parse(timetable.getStartTime());
+            Date end1 = format.parse(schedule.getEndTime());
+            Date end2 = format.parse(timetable.getEndTime());
+
+            if (start1.equals(start2) || end1.equals(end2)) {
+                return true;
+            }
+
+            if (start1.after(start2) && start1.before(end2)) {
+                return true;
+            }
+
+            if (start2.after(start1) && start2.before(end1)) {
+                return true;
+            }
```

Comment:
Format of start1 start2 is q confusing lol can use startToAdd and startExisting to prevent confusion?
Also can simplify check to just be if endToAdd.before(startExisting) | startToAdd.after(endExisting) = no clash